tests/rptest/tests/topic_recovery_test.py

```diff
@@ -176,17 +176,32 @@ def _validate_partition_last_offset(self):
 
     def _produce_and_verify(self, topic_spec):
         """Try to produce to the topic. The method produces data to the topic and
-        checks that high watermark advanced."""
+        checks that high watermark advanced. Wait for partition metadata to appear
+        in case of a recent leader election"""
+        def watermark_is_present(hwm, topic_name):
+            for partition_info in self._rpk.describe_topic(topic_name):
+                hwm = partition_info.high_watermark
```

Comment:
should we calculate the `max/min/first value which is not None` watermark over all partitions? 

right now we are effectively just getting the value from the last row that rpk produced. 

cc @Lazin

Reply:
We only need high watermark for this test.